### ISSUE TITLE
On Windows, can't create custom pssessionconfiguration against 6.0 endpoint

### ISSUE THREAD

**Issue by SteveL-MSFT** [PowerShell/PowerShell#4275]
`Register-PSSessionConfiguration` creates a new configuration but uses the microsoft.powershell endpoint which defaults to Windows PowerShell 5.1.

Steps to reproduce
------------------

```powershell
PS> New-PSSessionConfigurationFile -Path .\restricted.pssc -SessionType RestrictedRemoteServer -FunctionDefinitions @{Name="psversion";ScriptBlock={$psversiontable}}
PS> Register-PSSessionConfiguration -Path .\restricted.pssc -Name restricted

Type            Keys                                Name
----            ----                                ----
Container       {Name=restricted}                   restricted

PS> Restart-Service winrm
PS> Enter-PSSession -ComputerName . -ConfigurationName restricted
[localhost]: PS>psversion
```

Expected behavior
-----------------

```none
Name                           Value
----                           -----
PSVersion                      6.0.0-beta
PSEdition                      Core
GitCommitId                    v6.0.0-beta.4
OS                             Microsoft Windows 10.0.16245
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

Actual behavior
---------------

```none
Name                           Value
----                           -----
PSVersion                      5.1.16245.1000
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.16245.1000
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

Environment data
----------------

```powershell
PSVersion                      6.0.0-beta
PSEdition                      Core
GitCommitId                    v6.0.0-beta.4
OS                             Microsoft Windows 10.0.16245
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by mirichmo** [comment]
PR #2671 will probably fix it

**Comment by mirichmo** [comment]
Resolved via #2671
